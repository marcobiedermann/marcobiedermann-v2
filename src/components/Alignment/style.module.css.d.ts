export const alignment: string;
export const alignmentAlignCenter: string;
export const alignmentAlignEnd: string;
export const alignmentAlignStart: string;
export const alignmentJustifyCenter: string;
export const alignmentJustifyEnd: string;
export const alignmentJustifySpaceBetween: string;
export const alignmentJustifyStart: string;
