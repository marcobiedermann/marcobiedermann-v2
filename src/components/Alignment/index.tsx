import classNames from 'classnames';
import camelCase from 'lodash/camelCase';
import capitalize from 'lodash/capitalize';
import React, { FC } from 'react';
import * as styles from './style.module.css';

type Align = 'center' | 'end' | 'start';
type AlignClass = 'alignmentAlignCenter' | 'alignmentAlignEnd' | 'alignmentAlignStart';
type Justify = 'center' | 'end' | 'space-between' | 'start';
type JustifyClass =
  | 'alignmentJustifyCenter'
  | 'alignmentJustifyEnd'
  | 'alignmentJustifySpaceBetween'
  | 'alignmentJustifyStart';

export interface AlignmentProps {
  align?: Align;
  className?: string;
  justify?: Justify;
}

export const Alignment: FC<AlignmentProps> = (props) => {
  const { align, className, justify, ...otherProps } = props;

  return (
    <div
      className={classNames(className, {
        [styles.alignment]: align || justify,
        [styles[camelCase(`alignmentAlign${capitalize(align)}`) as AlignClass]]: align,
        [styles[camelCase(`alignmentJustify${capitalize(justify)}`) as JustifyClass]]: justify,
      })}
      {...otherProps}
    />
  );
};

export default Alignment;
