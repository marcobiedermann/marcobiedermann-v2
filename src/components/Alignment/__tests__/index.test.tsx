/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Alignment from '..';

describe('Alignment component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Alignment />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render content correctly', () => {
    const { asFragment } = render(<Alignment>Content</Alignment>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render align `center` correctly', () => {
    const { asFragment } = render(<Alignment align="center" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render align `end` correctly', () => {
    const { asFragment } = render(<Alignment align="end" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render align `start` correctly', () => {
    const { asFragment } = render(<Alignment align="start" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render justify `center` correctly', () => {
    const { asFragment } = render(<Alignment justify="center" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render justify `end` correctly', () => {
    const { asFragment } = render(<Alignment justify="end" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render justify `space-between` correctly', () => {
    const { asFragment } = render(<Alignment justify="space-between" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render justify `start` correctly', () => {
    const { asFragment } = render(<Alignment justify="start" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
