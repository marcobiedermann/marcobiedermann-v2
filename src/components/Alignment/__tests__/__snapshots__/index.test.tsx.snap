// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alignment component should render align \`center\` correctly 1`] = `
<div
  className="alignment alignmentAlignCenter"
/>
`;

exports[`Alignment component should render align \`end\` correctly 1`] = `
<div
  className="alignment alignmentAlignEnd"
/>
`;

exports[`Alignment component should render align \`start\` correctly 1`] = `
<div
  className="alignment alignmentAlignStart"
/>
`;

exports[`Alignment component should render content correctly 1`] = `
<div
  className=""
>
  Content
</div>
`;

exports[`Alignment component should render correctly 1`] = `
<div
  className=""
/>
`;

exports[`Alignment component should render justify \`center\` correctly 1`] = `
<div
  className="alignment alignmentJustifyCenter"
/>
`;

exports[`Alignment component should render justify \`end\` correctly 1`] = `
<div
  className="alignment alignmentJustifyEnd"
/>
`;

exports[`Alignment component should render justify \`space-between\` correctly 1`] = `
<div
  className="alignment alignmentJustifySpaceBetween"
/>
`;

exports[`Alignment component should render justify \`start\` correctly 1`] = `
<div
  className="alignment alignmentJustifyStart"
/>
`;
